### PR TITLE
mgr/dashboard_v2: Various fixes around return types and content types

### PR DESCRIPTION
**Pull request by p-se**
Unifies the output format and corresponding content type headers.

### CODE REVIEW

File: src/pybind/mgr/dashboard_v2/tools.py [hunk 1]
```diff
@@ -94,16 +95,13 @@ class RESTController(object):
...
-        'request.error_page': {'default': _json_error_page},
-    }
-
     def _not_implemented(self, is_element):
         methods = [method
                    for ((method, _is_element), (meth, _))
                    in self._method_mapping.items()
                    if _is_element == is_element and hasattr(self, meth)]
         cherrypy.response.headers['Allow'] = ','.join(methods)
+        cherrypy.response.headers['Content-Type'] = 'application/json'
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard_v2/tools.py]: do we still need this one?

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]

**Comment by rjfd** [comment]
@p-na to fix conflicts of PRs with respect to changes in the master branch (in this case `wip-mgr-dashboard_v2`) you should not merge the latest changes into the PR branch but instead rebase all the PR commits on top of the latest changes of master.

For example, if your git remote repo is called `ceph_openattic`, and assuming your current branch is the PR branch, you should do this:
```
# fetch latest changes from remote
git fetch ceph_openattic
# rebase 
git rebase ceph_openattic/wip-mgr-dashboard_v2
```

If conflicts arise during the rebase, do this:
```
# check which files are in conflict
git status
# edit the files and fix conflicts
git add files_with_fixed_conflicts
# continue the rebase
git rebase --continue
```

Please remove the merge commit and rebase this PR commits on top of the latest changes of `wip-mgr-dashboard_v2`

File: src/pybind/mgr/dashboard_v2/tools.py (hunk 1 above)
Comment (p-se) [reply src/pybind/mgr/dashboard_v2/tools.py]: Nope, nice find! Removed it.

**Review by sebastian-philipp — APPROVED** [review]

File: src/pybind/mgr/dashboard_v2/tools.py [hunk 2]
```diff
@@ -71,6 +71,7 @@ def load_controllers(mgrmodule):
 
 
 def _json_error_page(status, message, traceback, version):
```
Comment (votdev) [inline src/pybind/mgr/dashboard_v2/tools.py]: Maybe you should rename this function to ``json_error_page`` because it is not private anymore.

**Review by votdev — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/dashboard_v2/module.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 from mgr_module import MgrModule
 
 from .controllers.auth import Auth
-from .tools import load_controllers
+from .tools import load_controllers, _json_error_page
```
Comment (rjfd) [inline src/pybind/mgr/dashboard_v2/module.py]: please remove the `_` prefix from the function name since it's being used externally to the tools module

**Review by rjfd — CHANGES_REQUESTED** [review]

**Review by s0nea — APPROVED** [review]

**Review by rjfd — APPROVED** [review]
lgtm

**Review by votdev — APPROVED** [review]

**Pull request merged**